client/src/extension.ts

```diff
@@ -91,12 +94,93 @@ export function activate(context: ExtensionContext) {
 	}
 
 	// Create the language client and start the client.
-	let disposable = new LanguageClient('cfnLint', 'CloudFormation linter Language Server', serverOptions, clientOptions).start();
+	let languageClient = new LanguageClient('cfnLint', 'CloudFormation linter Language Server', serverOptions, clientOptions);
+	let disposable = languageClient.start();
+
+	languageClient.onReady().then(() => {
+		languageClient.onNotification('cfn/previewisavailable', (uri) => {
+			// This enables the "show preview" button in the top right corner
+			vscode.commands.executeCommand('setContext', 'isCfnPreviewAvailable', true);
+
+			// We don't want to show the preview if the user didn't request it. But, if the preview is visible, reload it
+			if (previews[uri]) {
+				vscode.commands.executeCommand('extension.sidePreview');
+			}
+		})
+		languageClient.onNotification('cfn/fileclosed', (uri) => {
+			// if the user closed the template itself
+			previews[uri].dispose();
+		})
+	})
 
 	// Push the disposable to the context's subscriptions so that the
 	// client can be deactivated on extension deactivation
 	context.subscriptions.push(disposable);
 
+	context.subscriptions.push(
+		vscode.commands.registerCommand('extension.sidePreview', loadSidePreview)
+	  );
+
+}
+
+function loadSidePreview() {
+	if (!window.activeTextEditor.document)
+		return;
+	let uri = window.activeTextEditor.document.uri;
+	//FIXME security concern - we should read this file from the extension's workspace
+	let dotFile = uri.fsPath + ".dot";
+
+	if (!fs.existsSync(dotFile)) {
+		//FIXME test!
+		window.showInformationMessage("Your version of cfn-python-lint doesn't support previews. Please update to the latest version.")
+		return;
+	}
+	const onDiskPath = vscode.Uri.file(dotFile);
+	const previewKey = uri.toString();
+
+	if (!previews[previewKey]) {
+		previews[previewKey] = vscode.window.createWebviewPanel(
+			'cfnLintPreview', // Identifies the type of the webview. Used internally
+			'Template: ' + dotFile, // Title of the panel displayed to the user
```

Comment:
thoughts on stripping the `.dot` from the panel title? Feels like an implementation detail
```suggestion
			'Template: ' + dotFile.slice(0,-4), // Title of the panel displayed to the user
```